### DISCUSSION TITLE
angular/how-to-get-current-route-in-angular/

### DISCUSSION THREAD

**Discussion by giscus** [arungudelli/angularjswiki#199] [Announcements]
# angular/how-to-get-current-route-in-angular/

Steps to get current route URL in Angular. 1. Import Router,NavigationEnd from angular/router and inject in the constructor. 2. Subscribe to the NavigationEnd event of the router.3. Get the current route Url by accessing NavigationEnd url property.

https://www.angularjswiki.com/angular/how-to-get-current-route-in-angular/

**Comment by chekinih** [comment]
Hi, Thank you.

I used this code to get things work:

    constructor(private router: Router) {}

    ngOnInit(): void {

      this.router.events.subscribe(event => {
        if (event instanceof NavigationEnd) {
          this.currentUrl = event.url;
        }
      });
    }
